src/ts/components/forms/slider.tsx

```diff
@@ -0,0 +1,358 @@
+    };
+  };
+
+  private convertRangeToPercentageRange(range: Range) {
+    const { min, max } = this.props;
+    return {
+      from: convertStepToPercentageStep(min, max, range.from),
+      to: convertStepToPercentageStep(min, max, range.to),
+    };
+  }
+
+  private convertPercentageRangeToRange(range: Range) {
+    const { min, max } = this.props;
+    return {
+      from: getClosestValue(
+        this.stepSeries,
+        convertPercentageStepToStep(min, max, range.from)
+      ),
+      to: getClosestValue(
+        this.stepSeries,
+        convertPercentageStepToStep(min, max, range.to)
+      ),
+    };
+  }
+
+  private createTouchHandlers(callback: (event: SyntheticTouchEvent) => void) {
+    function createSyntheticEvent(
+      event: TouchEvent | React.TouchEvent<HTMLElement>
+    ): SyntheticTouchEvent {
+      return {
+        type: event.type,
+        targetTouches: event.targetTouches,
+      };
+    }
+
+    const onTouchMove = (event: TouchEvent) => {
+      callback(createSyntheticEvent(event));
+    };
+
+    const onTouchEnd = (event: TouchEvent) => {
+      callback(createSyntheticEvent(event));
+
+      window.removeEventListener('touchend', onTouchEnd);
+      window.removeEventListener('touchmove', onTouchMove);
+    };
+
+    const onTouchDown = (event: React.TouchEvent<HTMLElement>) => {
+      callback(createSyntheticEvent(event));
+
+      window.addEventListener('touchend', onTouchEnd);
+      window.addEventListener('touchmove', onTouchMove);
+    };
+
+    this.unsubscribers.push(() => {
+      window.removeEventListener('touchmove', onTouchMove);
+      window.removeEventListener('touchend', onTouchEnd);
+    });
+
+    return onTouchDown;
+  }
+
+  private createMouseHandlers(callback: (event: SyntheticMouseEvent) => void) {
+    function createSyntheticEvent(
+      event: MouseEvent | React.MouseEvent<HTMLElement>
+    ): SyntheticMouseEvent {
+      return {
+        type: event.type,
+        clientX: event.clientX,
+      };
+    }
+
+    const onMouseMove = (event: MouseEvent) => {
+      callback(createSyntheticEvent(event));
+    };
+
+    const onMouseUp = (event: MouseEvent) => {
+      callback(createSyntheticEvent(event));
+
+      window.removeEventListener('mouseup', onMouseUp);
+      window.removeEventListener('mousemove', onMouseMove);
+    };
+
+    const onMouseDown = (event: React.MouseEvent<HTMLElement>) => {
+      if (event.button === 2) {
+        return;
+      }
+
+      event.preventDefault();
+
+      callback(createSyntheticEvent(event));
+
+      window.addEventListener('mouseup', onMouseUp);
+      window.addEventListener('mousemove', onMouseMove);
+    };
+
+    this.unsubscribers.push(() => {
+      window.removeEventListener('mousemove', onMouseMove);
+      window.removeEventListener('mouseup', onMouseUp);
+    });
+
+    return onMouseDown;
+  }
+
+  private onHandleFrom = (event: SyntheticMouseEvent | SyntheticTouchEvent) => {
+    this.setState({
```

Comment:
You could add a `this.props.onSlide` call in here (optional) so we can track as the values are being changed also. Not sure if necessary. Same for `onHandleTo`.

Reply:
This is actually what `onChange` is tracking right?
Not sure what another `onSlide` would add.

I do now realise that a lot of the code in `onHandleFrom` is the same as in `onHandleTo` though, so will abstract that out into separate function

Reply:
Have changed `onChange` to `onSlide` after convo on Slack